### ISSUE TITLE
sugar syntax to declaring views

### ISSUE THREAD

**Issue by sibelius** [Automattic/mongoose#7633]
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
a mongoose syntax to create views

**What is the expected behavior?**
a mongoose syntax to create views

it could be similar to mongoose.Schema

maybe

```jsx
mongoose.View({
   name: '',
   viewOn: '',
   pipeline: [],
});
```

**Comment by vkarpov15** [comment]
Will think about what we can do with views. Any more detailed suggestions are welcome

**Comment by 3axap4eHko** [comment]
What do you think about creating a model with additional schema parameters to be consistent with MongoDB doc
```
export const Schema = new Mongoose.Schema({

}, { collection: '<collection>', viewOn: '<source>', pipeline: '<pipeline>', collation: '<collation>' });

export default connection.model('View', Schema);
```
Also, such models should be marked as read-only

**Comment by sibelius** [comment]
does this also apply to On Demand Materialized Views?

https://www.mongodb.com/docs/manual/core/materialized-views/ ?

view here https://www.mongodb.com/docs/manual/core/views/

**Comment by vkarpov15** [comment]
@sibelius yeah, from a Mongoose models perspective materialized views and standard views are equivalent.

**Comment by sibelius** [comment]
what is the best approach to create an on-demand materialized view on mongoose?

should we fallback to mongodb?

or is this approach a valid one https://www.hownowbrownkow.com/posts/mongodb-materialized-views?

**Comment by vkarpov15** [comment]
https://www.hownowbrownkow.com/posts/mongodb-materialized-views looks valid. Create a Mongoose model on top of your view, and use that model like you would any other model.

**Comment by sibelius** [comment]
good enough for now

**Comment by vkarpov15** [comment]
@sibelius I'm going to leave this open for future dev work, if you don't mind
